core/string/ustring.cpp

```diff
@@ -4280,12 +4274,12 @@ String String::pad_decimals(int p_digits) const {
 		c = s.length() - 1;
 	} else {
 		if (p_digits <= 0) {
-			return s.substr(0, c);
+			return s.left(c);
```

Comment:
`c` is the result of `find()`, and this is the case where it's not -1, so it's safe.